src/reduct/api/entry_api.cc

```diff
@@ -175,12 +175,14 @@ Result<HttpRequestReceiver> EntryApi::Read(IStorage* storage, std::string_view b
                 .headers = std::move(headers),
                 .content_length = reader->size(),
                 .SendData =
-                    [reader]() {
-                      auto [chunk, err] = reader->Read();
-                      return Result<std::string>{std::move(chunk.data), err};
+                    [reader, send_record_data]() {
```

Comment:
this is completely fine to pass "send_record_data into the lambda. Consider using the flag to create two different lambdas.